### PR TITLE
Library Challenge 2019

### PR DESCRIPTION
**Pull request by EevanR**
Library Challenge 2019

PR content
In this PR we are submitting our library program.
This PR contains

ability to create a user
ability to initialize a library
On library initialize a group of books is loaded from a YAML file
README

What We have learned

Continuous striving to become more proficient with rspec and creating the proper implementation
more work with classes
using an initialize method to set up the library books database
more work with instance methods
using instance double to preform testing
refactoring for easier readability
creating a properly structured README with proper markdown
Authors

Johan S., Eevan R.

### CODE REVIEW

File: spec/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require './lib/user.rb'
+require 'date'
+
+describe User do
+
+    it 'creates a user with no checked-out books' do
+        expect(subject.books_in_posession).to be_empty
+    end
+
+    #it 'populates books posession list with newly checked out books' do
```
Comment (oliverochman) [inline spec/user_spec.rb]: please remove commented code

File: spec/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    #    expect(subject.borrowed_books).to eq 
+    #end
+
+    let(:library) { Library.new }
+    it 'checks out book from library, storing its title, author and return date' do
+        expected_output = ["Alfons och soldatpappan - Gunilla Bergström (Return date: #{Date.today.next_month(1).strftime("%Y-%m-%d")})"]
+        expect(subject.checkout_book(library, "Alfons och soldatpappan")).to eq expected_output
```
Comment (oliverochman) [inline spec/user_spec.rb]: I would like to see an it block where u test the `@books_in_possession` instead of the response from `checkout_book`

**Review by oliverochman — APPROVED** [review]

File: spec/user_spec.rb (hunk 1 above)
Comment (EevanR) [reply spec/user_spec.rb]: done

File: spec/user_spec.rb (hunk 2 above)
Comment (EevanR) [reply spec/user_spec.rb]: done

**Pull request closed without merging**